src/Symfony/Component/Validator/Constraints/NoBannedWords.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * @Annotation
+ *
+ * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+final class NoBannedWords extends Constraint
+{
+    public const BANNED_WORDS_ERROR = 'd187ff45-bf23-4331-aa87-c24a36e9b400';
+
+    protected const ERROR_NAMES = [
+        self::BANNED_WORDS_ERROR => 'BANNED_WORDS_ERROR',
+    ];
+
+    public string $message = 'The value contains the following banned words: {{ wordList }}.';
+
+    /**
+     * @var array<string>
+     */
+    public array $dictionary = [];
```

Comment:
so, same question: shouldn't the dictionary be configured at the validator level, via its constructor? how is one supposed to declare this list to the constraint if it's a ling one?

Reply:
Actually it could be used set either from the constraint itself for specific cases (e.g. initially with user data) or globally from a big dictionnary (e.g. forum post checks).

Reply:
Let see this in the dedicated PR https://github.com/symfony/symfony/pull/49868